app/src/main/report/reducers/reportsReducer.ts

```diff
@@ -62,6 +65,30 @@ export const reportsReducer = (state = reportsInitialState, action: ReportsActio
         case ReportTypeKeys.REPORT_UNPUBLISHED:
             report = (action as ReportUnpublished).data;
             return getUpdatePublishedState(report, false)
+        case ReportTypeKeys.REPORT_RUN_STARTED:
+            let runningReports = state.runningReports.filter(r => r.name != action.data.name);
+            runningReports.push({
+                name: action.data.name,
+                key: action.data.key,
+                status: RunningReportStatusValues.RUNNING_REPORT_STATUS_STARTED,
+                version: null,
+                output: null
+            });
+            return {...state, runningReports: runningReports};
+        case ReportTypeKeys.REPORT_RUN_STATUS_FETCHED:
+            runningReports = state.runningReports;
+            const runningReport = runningReports.find(r => r.key == action.data.key);
+            //Update the object in the state
```

Comment:
We shouldn't actually update state objects, only replace: https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns